ui/src/shared/actions/sources.ts

```diff
@@ -200,14 +203,86 @@ export const deleteKapacitorAsync = (kapacitor: Kapacitor) => async (
   }
 }
 
-export const getSourcesAsync = () => async (dispatch): Promise<void> => {
+export const getSourcesAsync = () => async (dispatch): Promise<Source[]> => {
   try {
-    const {
-      data: {sources},
-    } = await getSourcesAJAX()
-    dispatch(loadSources(sources))
-    return sources
+    const {data} = await getSourcesAJAX()
+
+    const sources: Source[] = data.sources
+
+    const sourcesWithFluxLink = await Promise.all(sources.map(addSourceLink))
+
+    dispatch(loadSources(sourcesWithFluxLink))
+    return sourcesWithFluxLink
   } catch (error) {
     dispatch(errorThrown(error))
   }
 }
+
+const addSourceLink = async (source: Source): Promise<Source> => {
+  try {
+    const services = await getServicesAJAX(source.links.services)
+    const fluxServices = services.filter(s => s.type === 'flux')
+
+    let service: Service
+
+    if (validateSupportsFlux(source)) {
+      if (!fluxServices.length) {
+        service = await createFluxServiceForSource(source)
+      } else {
+        if (fluxServices[0].url !== source.url) {
+          service = await updateFluxService(source, fluxServices[0])
+        } else {
+          service = fluxServices[0]
+        }
+      }
+
+      const {links} = source
+      const linksWithFlux: SourceLinks = {...links, flux: service.links.proxy}
+
+      return {...source, links: linksWithFlux}
+    }
+
+    return source
+  } catch (err) {
+    return source
+  }
+}
+
+const createFluxServiceForSource = async (source: Source) => {
+  const {url} = source
+
+  const service: NewService = {
+    name: 'Flux',
+    url,
+    username: '',
+    insecureSkipVerify: false,
+    type: 'flux',
+    metadata: {
+      active: true,
+    },
+  }
+  const s = await createServiceAJAX(source, service)
+  return s
+}
+
+const updateFluxService = async (source: Source, service: Service) => {
+  const {url} = source
+  const updatedService = {...service, url}
+  const s = await updateServiceAJAX(updatedService)
+  return s
+}
+
+const validateSupportsFlux = (source: Source) => {
+  if (source.type !== 'influx') {
```

Comment:
what are the other types for a source?

Reply:
i believe 'influx-enterprise' and maybe more, but we don't want to do this for those.